src/ol/controls/RoutingControl.js

```diff
@@ -0,0 +1,475 @@
+} from '@turf/helpers';
+import RoutingAPI from '../../api/routing/RoutingAPI';
+import Control from '../../common/controls/Control';
+import RoutingLayer from '../layers/RoutingLayer';
+
+/**
+ * Display a route of a specified mean of transport.
+ *
+ * @example
+ * import { Map, RoutingControl } from 'mobility-toolbox-js/ol';
+ *
+ * const map = new Map({
+ *   target: 'map',
+ *   controls: [
+ *     new RoutingControl()
+ *   ]
+ * });
+ *
+ *
+ * @see <a href="/example/ol-routing">Openlayers routing example</a>
+ *
+ * @extends {Control}
+ * @implements {RoutingInterface}
+ */
+class RoutingControl extends Control {
+  constructor(options = {}) {
+    super(options);
+
+    Object.defineProperties(this, {
+      mot: {
+        get: () => {
+          return this.get('mot');
+        },
+        set: (newMot) => {
+          if (newMot) {
+            this.set('mot', newMot);
+            if (this.viaPoints) {
+              this.drawRoute();
+            }
+          }
+        },
+      },
+    });
+
+    this.mot = options.mot || 'bus';
+
+    this.abortController = new AbortController();
+
+    this.apiKey = options.apiKey;
+
+    this.api = new RoutingAPI({
+      url: options.url,
+      apiKey: this.apiKey,
+      mot: options.mot,
+    });
+
+    this.routingLayer =
+      options.routingLayer ||
+      new RoutingLayer({
+        name: 'routing-layer',
+      });
+
+    this.onRouteError =
+      options.onRouteError ||
+      ((error) => {
+        this.dispatchEvent({
+          type: 'change:route',
+          target: this,
+        });
+        this.viaPoints = [];
+        this.routingLayer.olLayer.getSource().clear();
+        // eslint-disable-next-line no-console
+        console.error(error);
+      });
+
+    this.viaPoints = [];
+    this.setDrawEnabled = this.setDrawEnabled.bind(this);
+    this.onMapClick = this.onMapClick.bind(this);
+    this.onModifyEnd = this.onModifyEnd.bind(this);
+    this.onModifyStart = this.onModifyStart.bind(this);
+    this.apiChangeListener = () => this.drawRoute();
+    this.createModifyInteraction();
+  }
+
+  /**
+   * Enables/Disables route draw mode.
+   * @param {boolean} enabled Enable/Disable route draw mode
+   * @param {boolean} [reset=undefined] Clears all features and viaPoints
+   */
+  setDrawEnabled(enabled, reset) {
+    if (this.map) {
+      if (reset) {
+        // Clear viaPoints and source
+        this.dispatchEvent({
+          type: 'change:route',
+          target: this,
+        });
+        this.viaPoints = [];
+        this.routingLayer.olLayer.getSource().clear();
+      }
+
+      if (!enabled) {
+        // Deactivate modify interaction and remove listeners
+        this.modifyInteraction.setActive(false);
+        this.removeListeners();
+        return;
+      }
+
+      // Activate modify interaction and add listeners
+      this.map.addLayer(this.routingLayer);
+      this.modifyInteraction.setActive(true);
+      this.addListeners();
```

Comment:
you can move all of this in activate and deactivate functions.

Reply:
@oterral setDrawEnabled was defined by Mario, which is why I added it. But it probably makes sense moving everything into activate/deactivate and removing it